### ISSUE TITLE
import '.' is changed to import './'

### ISSUE THREAD

**Issue by neodon** [daidodo/tsimportsorter#28]
First, thank you for taking over this extension and improving it.  I was sorry to see [TypeScript Import Sorter](https://marketplace.visualstudio.com/items?itemName=mike-co.import-sorter) become unmaintained.  I'm glad to have an up-to-date alternative because it saves my team a lot of hassle and code conflicts with ordering of imports.

**Describe the bug**
When I need to import the default module in a folder from another module in the same folder, I typically do it with:

```javascript
import foo from '.'
```

This extension automatically replaces that import with:

```javascript
import foo from './'
```

Importing from `'./'` (and even `'./index'`) triggers linting errors due to it being redundant (`'.'` is sufficient):

```
  ✖  2:22  Useless path segments for "./", should be "."                   import/no-useless-path-segments
```

Note: This happens with both JavaScript and TypeScript.

**To Reproduce**
1. Create a file in `src` named `index.js`.  In `index.js`, put this code:
```javascript
const foo = 'bar'
export default foo
```
2. Create another file in `src` named `cli.js`.  In `cli.js`, put this code:
```javascript
import foo from '.'
console.log(foo)
```
3. Save `foo.js` or otherwise trigger the import sorting extension.
4. See the import path change.

**Expected behavior**
I expect the import path to be the shortest/simplest path necessary to successfully import the module.  In this case, it should stay as `'.'`.

**Actual behavior**
The import path is always changed to `'./'`, which triggers linting errors for redundancy in the path.

**Screenshots**

**OS (please complete the following information):**
 - OS: Windows
 - Version: 10

**VS Code (please complete the following information):**
- Version: 1.51.1 (latest stable as of 11/15/2020)

**repo with reproduction of issue**
https://github.com/neodon/tsimportsorter-issue

**Installed VS Code extensions**
I disabled all extensions except JS/TS Import/Export Sorter v3.0.4

I appreciate any discussion, and if there is agreement that this is a bug, I would be interested in submitting a pull request.  Thank you.

Edit: After looking at the tests for path normalization and giving it some more thought, I think the fix for this bug would be limited to paths like `.`, `..`, `../..`, etc and not something like `./foo/`.  The reason is that you might have both `foo.js` and `foo/index.js`, so the trailing slash in those cases is needed to differentiate.

**Comment by daidodo** [comment]
@neodon 

Thanks for your feedback!

I think this one is duplicated to https://github.com/daidodo/tsimportsorter/issues/18 (Closed). But I'm happy to take any arguments since many people seem to ask the same question.

**Comment by neodon** [comment]
Thank you for the response.  I read through the discussions.  I'll echo the sentiment that it would be great to have an option to disable this behavior.  It seems that it's only necessary to support an anti-pattern of having conflicting package imports (e.g. `foo.js` + `foo/index.js` both exist).  I've never seen this in the wild (though I'm sure a few people do it) and have only encountered it once myself when I made an obvious mistake of converting a module file into a directory with an index.js and forgot to delete the old module file.

I'd rather the extension at least let me set an option so it does not "correct" my code by making it worse in order to support a (likely rare) anti-pattern / abuse of module paths that I would never use anyway.

**Comment by daidodo** [comment]
Path normalisation is necessary for merging imports and exports, so we can format:
```js
import { A } from "./a";
import { B } from  "././a";
import { C } from ".";
import { D } from "./.";
```
to
```js
import { A, B } from "./a";
import { C, D } from "./";
```
The only question is whether to keep the final `/` for `.`, `..`, etc.

I agree that we can add an option for `.`, `..`, etc, that:
* Always add the final slash, or
* Always remove the final slash.

At a first glance I think the core logic would be in [normalizePath](https://github.com/daidodo/tsimportsorter/blob/master/src/format/utils/index.ts#L4), along with other code for a new configuration.

I'd be appreciative if you could pick it up and help improve the extension.
Thanks!

**Comment by neodon** [comment]
Thank you! I appreciate the openness and guidance.  I will see what I can do.

**Comment by daidodo** [comment]
Hi @neodon,

Have you started working on this issue?

This issue is the next priority of my current project [eslint-plugin-import Compatibility](https://github.com/daidodo/tsimportsorter/projects/2). So I'm happy to work on it if you haven't started.

Thanks!

**Comment by neodon** [comment]
I've only updated some tests, so feel free to take it in another direction.  Here is the branch: https://github.com/neodon/tsimportsorter/tree/issue-28-minimize-paths

**Comment by daidodo** [comment]
@neodon, thanks for sharing!

This is a community project so please feel no pressure.

And if you like you can pick up any items from [eslint-plugin-import Compatibility](https://github.com/daidodo/tsimportsorter/projects/2) or [TODO](https://github.com/daidodo/tsimportsorter/blob/master/TODO.md) list.
The [CONTRIBUTING.md](https://github.com/daidodo/tsimportsorter/blob/master/CONTRIBUTING.md) could be a good start point in case you haven't noticed.

And it's also ok if you just want to use the extension and have fun!

Cheers!

**Comment by daidodo** [comment]
Hi @neodon, this issue should be resolved by the latest release v5.0.0.

Now you can set `removeLastSlashInPath` and `removeLastIndexInPath` in your config if you'd like useless `/` and `index` to be removed.

Furthermore, you can just set up ESLint to enable [import/no-useless-path-segments](https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-useless-path-segments.md) which has the same effect.

See also:
[Release notes](https://github.com/daidodo/tsimportsorter#50-release-notes)
[Wiki about how `no-useless-path-segments` is handled](https://github.com/daidodo/tsimportsorter/wiki/ESLint-Compatibility#importno-useless-path-segments)

Thanks!
